lib/src/widgets/confirm_recover_card.dart

```diff
@@ -0,0 +1,207 @@
+import 'dart:math';
+
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:font_awesome_flutter/font_awesome_flutter.dart';
+import 'animated_button.dart';
+import 'animated_text_form_field.dart';
+import '../models/login_data.dart';
+import '../providers/auth.dart';
+import '../providers/login_messages.dart';
+import '../widget_helper.dart';
+
+class ConfirmRecoverCard extends StatefulWidget {
+  ConfirmRecoverCard({
+    Key? key,
+    required this.passwordValidator,
+    required this.onBack,
+    required this.onSubmitCompleted,
+  }) : super(key: key);
+
+  final FormFieldValidator<String> passwordValidator;
+  final VoidCallback onBack;
+  final VoidCallback onSubmitCompleted;
+
+  @override
+  ConfirmRecoverCardState createState() => ConfirmRecoverCardState();
+}
+
+class ConfirmRecoverCardState extends State<ConfirmRecoverCard>
+    with SingleTickerProviderStateMixin {
+  final GlobalKey<FormState> _formRecoverKey = GlobalKey();
+
+  final _passwordFocusNode = FocusNode();
+  final _confirmPasswordFocusNode = FocusNode();
+  final _passwordController = TextEditingController();
+
+  var _isSubmitting = false;
+  var _code = '';
+
+  late AnimationController _submitController;
+
+  @override
+  void initState() {
+    super.initState();
+
+    _submitController = AnimationController(
+      vsync: this,
+      duration: Duration(milliseconds: 1000),
+    );
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+    _submitController.dispose();
+  }
+
+  Future<bool> _submit() async {
+    FocusScope.of(context).requestFocus(FocusNode()); // close keyboard
+
+    if (!_formRecoverKey.currentState!.validate()) {
+      return false;
+    }
+    final auth = Provider.of<Auth>(context, listen: false);
+    final messages = Provider.of<LoginMessages>(context, listen: false);
+
+    _formRecoverKey.currentState!.save();
+    await _submitController.forward();
+    setState(() => _isSubmitting = true);
+    final error = await auth.onConfirmRecover!(
+      _code,
+      LoginData(
+        name: auth.email,
+        password: auth.password,
+      ),
+    );
+
+    if (error != null) {
+      showErrorToast(context, null, error);
+      setState(() => _isSubmitting = false);
+      await _submitController.reverse();
+      return false;
+    }
+
+    showSuccessToast(context, null, messages.confirmRecoverSuccess);
+    setState(() => _isSubmitting = false);
+    widget.onSubmitCompleted();
```

Comment:
I don't think an account recovery should automatically login the user. Just like the 'normal' recovery, we should go back to the login screen.